api-specs/Gateway-EE/3.5/kong-ee-3.5.yaml

```diff
@@ -10156,88 +11210,2472 @@ paths:
+      tags:
+        - filter-chains
+      description: |
+        Update Filter Chain Associated to a Specific Route
+      requestBody:
+        $ref: '#/components/requestBodies/filter-chains-request'
+  '/services/{service_id_or_name}/filter-chains':
+    parameters:
+      - $ref: '#/components/parameters/service_id_or_name'
+    post:
+      summary: Create Filter Chain Associated to a Specific Service
+      tags:
+        - filter-chains
+      responses:
+        '201':
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Filter-chains'
+      operationId: get-services-service_id_or_name-filter-chains
+      description: |
+        Add filter chain Associated to a Specific Service
+      requestBody:
+        $ref: '#/components/requestBodies/filter-chains-request'
+    get:
+      summary: List Filter Chains Associated to a Specific Service
+      operationId: get-service_id_or_name-filter-chains
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Filter-chains'
+      description: |
+        List Filter Chains Associated to a Specific Service
+      tags:
+        - filter-chains
+  '/filter-chains/{filter_chain_id}':
+    parameters:
+      - $ref: '#/components/parameters/filter_chain_id'
+    get:
+      summary: Retrieve Filter Chain
+      tags:
+        - filter-chains
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Filter-chains'
+      operationId: get-filter-chains-filter_chain_id
+      description: |
+        Retrieve Filter Chain
+    patch:
+      summary: Update Filter Chain
+      operationId: patch-filter-chains-filter_chain_id
+      responses:
+        '201':
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Filter-chains'
+      tags:
+        - filter-chains
+      description: |
+        Update Filter Chain
+        Note: This API is not available in DB-less mode.
+      requestBody:
+        $ref: '#/components/requestBodies/filter-chains-request'
+    put:
+      summary: Update Or Create Filter Chain
+      operationId: put-filter-chains-filter_chain_id
+      responses:
+        '200':
+          description: OK
+      tags:
+        - filter-chains
+      description: |-
+        Inserts (or replaces) the Filter Chain under the requested resource with the definition specified in the body. The Filter Chain will be identified via the name or id attribute.
+
+        When the name or id attribute has the structure of a UUID, the Filter Chain being inserted/replaced will be identified by its id. Otherwise it will be identified by its name.
```

Comment:
```suggestion
        When the name or ID attribute has the structure of a UUID, the filter chain being inserted or replaced is identified by its ID. Otherwise, it is identified by its name.
```